### DISCUSSION TITLE
Model predicting null class at edges of chips

### DISCUSSION THREAD

**Discussion by NoaMills-USDA** [azavea/raster-vision#1922] [Q&A] [answered]
I've been tinkering with the quickstart model, and I get null pixel predictions in areas where data is indeed present. When I run the quickstart as is, I get null predictions on the borders of the validation raster where the chips don't reach. I made one change to the quickstart: I changed the pixel size from 300 to 325 so the prediction covers the entire area of the validation raster. Now, I get null pixels predicted in the middle of the raster, especially at the corners of the chips. Why is this? I'm only training for one epoch, but I figured there shouldn't be any guesswork in which pixels have no data.

Quickstart validation prediction:
![image](https://github.com/azavea/raster-vision/assets/100314908/b9423f98-960c-4adf-837f-24379ecf81e7)

Quickstart validation prediction after changing chip size to 325:
![image](https://github.com/azavea/raster-vision/assets/100314908/2439b338-2c4e-43fe-81c8-af2b64c894e4)

**Comment by AdeelH** [comment, marked as answer]
To the model, the null class is just like any other class. If you're only training for 1 epoch, it is likely that the model has just not learned to recognize no-data pixels yet. 

The null class pixels in the prediction are from two sources: 
- the model's predictions
- [RV automatically setting no-data pixels to the null class](https://github.com/azavea/raster-vision/blob/30fd26fc1c21aaf8e4f9414ce3a88dfda74539be/rastervision_core/rastervision/core/rv_pipeline/semantic_segmentation.py#L138-L145)
